README.md

```diff
@@ -1,128 +1,213 @@
 
-var client = new RealtimeClient(process.env.REALTIME_URL)
-client.connect()
-```
+# Usage
 
-REALTIME_URL is `'ws://localhost:4000/socket'` when developing locally and `'wss://<project_ref>.supabase.co/realtime/v1'` when connecting to your Supabase project.
+## Installing the Package
 
-You can pass in your JWT If you have enabled JWT authorization in Supabase [Realtime](https://github.com/supabase/realtime) server.
+```bash
+npm install @supabase/realtime-js
+```
+
+## Creating a Channel
 
 ```js
 import { RealtimeClient } from '@supabase/realtime-js'
 
-var client = new RealtimeClient(process.env.REALTIME_URL, {
-  params: { apikey: 'token123' },
+const client = new RealtimeClient(REALTIME_URL, {
+  params: {
+    apikey: API_KEY,
+    eventsPerSecond: 10,
+  },
 })
-client.connect()
-```
 
-See [Realtime: Websocket Connection Authorization](https://github.com/supabase/realtime#websocket-connection-authorization) for more information.
+const channel = client.channel('test-channel', {})
 
-### Subscribing to events
+channel.subscribe((status, err) => {
+  if (status === 'SUBSCRIBED') {
+    console.log('Connected!')
+  }
 
-You can listen to `INSERT`, `UPDATE`, `DELETE`, or all `*` events.
+  if (status === 'CHANNEL_ERROR') {
+    console.log(`There was an error subscribing to channel: ${err.message}`)
+  }
 
-You can subscribe to events on the whole database, schema, table, or individual columns using `channel()`. Channels are multiplexed over the Socket connection.
+  if (status === 'TIMED_OUT') {
+    console.log('Realtime server did not respond in time.')
+  }
 
-To join a channel, you must provide the `topic`, where a topic is either:
+  if (status === 'CLOSED') {
+    console.log('Realtime channel was unexpectedly closed.')
+  }
+})
+```
+
+### Notes:
 
-- `realtime` - entire database
-- `realtime:{schema}` - where `{schema}` is the Postgres Schema
-- `realtime:{schema}:{table}` - where `{table}` is the Postgres table name
-- `realtime:{schema}:{table}:{col}=eq.{val}` - where `{col}` is the column name, and `{val}` is the value which you want to match
+- `REALTIME_URL` is `'ws://localhost:4000/socket'` when developing locally and `'wss://<project_ref>.supabase.co/realtime/v1'` when connecting to your Supabase project.
+- `API_KEY` is a JWT whose claims must contain `exp` and `role` (existing database role).
+- Channel name can be any `string`.
+- `eventsPerSecond`, or client-side rate limiting, enforces the number of events sent to the Realtime server uniformly spread across a second. The default is 10, which means that the client can send one event, whether that's **Broadcast**/**Presence**/**Postgres CDC**, every 100 milliseconds. You may change this as you see fit but note that the server's rate limiting will need to be updated accordingly. You can learn more about Realtime's rate limits here: https://supabase.com/docs/guides/realtime/rate-limits.
 
-**Examples**
+### Broadcast
```

Comment:
nit: Should this be an h2 instead (along with all the following headings that aren't notes)? Assuming the notes are specific to the preceding heading...

Example
```
- Installing the package
  - Notes
- Broadcast
  - Notes
- Presence
- Postgres CDC
...
```

Reply:
@dannykng yep, good call. updated!